### PR TITLE
chore: `nargo info` now prints information as a prettified table 

### PR DESCRIPTION
**Pull request by Ethan-000**
# Description

## Problem\*

Related to https://github.com/noir-lang/noir/issues/2089

## Summary\*

run `nargo info`

```rust
fn main(x : Field, y : pub Field) {
    assert(x * 2 == y * 3);
}

contract Foo {
    fn double(x: Field) -> pub Field { x * 2 }
    fn triple(x: Field) -> pub Field { x * 3 }
    internal fn quadruple(x: Field) -> pub Field { x * 4 }
    open internal fn skibbidy(x: Field) -> pub Field { x * 5 }
}

```

```rust
+----------+-----------+------------------------+--------------+----------------------+
| Contract | Function  | Language               | ACIR Opcodes | Backend Circuit Size |
+----------+-----------+------------------------+--------------+----------------------+
| Foo      | quadruple | PLONKCSat { width: 3 } | 1            | 2                    |
+----------+-----------+------------------------+--------------+----------------------+
| Foo      | double    | PLONKCSat { width: 3 } | 1            | 2                    |
+----------+-----------+------------------------+--------------+----------------------+
| Foo      | triple    | PLONKCSat { width: 3 } | 1            | 2                    |
+----------+-----------+------------------------+--------------+----------------------+
| Foo      | skibbidy  | PLONKCSat { width: 3 } | 1            | 1                    |
+----------+-----------+------------------------+--------------+----------------------+

```

## Additional Context

# PR Checklist\*

- [x] I have tested the changes locally.
- [x] I have formatted the changes with [Prettier](https://prettier.io/) and/or `cargo fmt` on default settings.

### CODE REVIEW

File: crates/nargo_cli/build.rs [hunk 1]
```diff
@@ -110,7 +110,7 @@ fn compile_success_{test_name}() {{
     cmd.arg("info");
 
     // `compile_success` tests should be able to compile down to an empty circuit.
-    cmd.assert().stdout(predicate::str::contains("Total ACIR opcodes generated for language PLONKCSat {{ width: 3 }}: 0"));
+    cmd.assert().stdout(predicate::str::contains("ACIR Opcodes | Backend Circuit Size"));
```
Comment (TomAFrench) [inline crates/nargo_cli/build.rs:113]: This removes the point of this integration test which shows that the program is optimized down to an empty circuit.

**Review by TomAFrench — CHANGES_REQUESTED** [review]
Table looks good but you've made the `compile_success` integration tests must less strict.

**Comment by TomAFrench** [comment]
Will review on Monday. Work life balance ser

**Review by TomAFrench — APPROVED** [review]

**Comment by TomAFrench** [comment]
Currently if you use this in a workspace you get the output

```
+---------+------------------------+--------------+----------------------+
| Package | Language               | ACIR Opcodes | Backend Circuit Size |
+---------+------------------------+--------------+----------------------+
| a       | PLONKCSat { width: 3 } | 5            | 5                    |
+---------+------------------------+--------------+----------------------+
+---------+------------------------+--------------+----------------------+
| Package | Language               | ACIR Opcodes | Backend Circuit Size |
+---------+------------------------+--------------+----------------------+
| b       | PLONKCSat { width: 3 } | 5            | 5                    |
+---------+------------------------+--------------+----------------------+
```

Ideally this would be merged as so:

```
+---------+------------------------+--------------+----------------------+
| Package | Language               | ACIR Opcodes | Backend Circuit Size |
+---------+------------------------+--------------+----------------------+
| a       | PLONKCSat { width: 3 } | 5            | 5                    |
+---------+------------------------+--------------+----------------------+
| b       | PLONKCSat { width: 3 } | 5            | 5                    |
+---------+------------------------+--------------+----------------------+
```

Similarly we should collect contracts across different packages and collate them into the same table.

**Review by TomAFrench — CHANGES_REQUESTED** [review]
See comment

**Comment by kevaundray** [comment]
> Currently if you use this in a workspace you get the output
> 
> ```
> +---------+------------------------+--------------+----------------------+
> | Package | Language               | ACIR Opcodes | Backend Circuit Size |
> +---------+------------------------+--------------+----------------------+
> | a       | PLONKCSat { width: 3 } | 5            | 5                    |
> +---------+------------------------+--------------+----------------------+
> +---------+------------------------+--------------+----------------------+
> | Package | Language               | ACIR Opcodes | Backend Circuit Size |
> +---------+------------------------+--------------+----------------------+
> | b       | PLONKCSat { width: 3 } | 5            | 5                    |
> +---------+------------------------+--------------+----------------------+
> ```
> 
> Ideally this would be merged as so:
> 
> ```
> +---------+------------------------+--------------+----------------------+
> | Package | Language               | ACIR Opcodes | Backend Circuit Size |
> +---------+------------------------+--------------+----------------------+
> | a       | PLONKCSat { width: 3 } | 5            | 5                    |
> +---------+------------------------+--------------+----------------------+
> | b       | PLONKCSat { width: 3 } | 5            | 5                    |
> +---------+------------------------+--------------+----------------------+
> ```
> 
> Similarly we should collect contracts across different packages and collate them into the same table.

Why do we want this to be merged?

**Comment by kevaundray** [comment]
> now print something like this 😂
> with color

How does this print for contracts?

**Comment by kevaundray** [comment]
> I suppose depends on whether in a workspace, can different packages be compiled to different languages?

I think its a matter of preference, so rationale for preferences would be good

**Review by TomAFrench — APPROVED** [review]
LGTM

**Comment by kevaundray** [comment]
oof sorry @TomAFrench seems there was a race condition with changing the PR title 😁

**Pull request merged**